File: web/src/lib/server/api/interface/home/quick-picks.ts
import type { QuickPickMusic } from "../types.js";
import type { CommonOptions } from "./types.js";

import { DEFAULT_REGION } from "$lib/constants/countries.js";
import fetchFromRapidApi from "../rapid-api-fetcher.js";

export interface QuickPicksResponse {
  error: boolean;
  results: QuickPickMusic[];
}

export default async function getQuickPicks({ gl = DEFAULT_REGION }: CommonOptions) {
  const response = await fetchFromRapidApi<QuickPicksResponse>("/recommend", {
    searchParams: { gl },
  });

  return response;
}

File: server/api/interface/quick_picks.go
package api_interface

import (
	"encoding/json"
	"wavelength/constants"
	api_models "wavelength/models/api"
)

func (*YouTubeMusicClient) GetQuickPicks(gl string) (*api_models.QuickPicksResponse, error) {
	if gl == "" {
		gl = constants.DefaultRegion
	}

	response, err := rapidApiFetch("/recommend", map[string]string{
		"gl": gl,
	})

	if err != nil {
		return nil, err
	}

	defer response.Body.Close()

	var quickPicks api_models.QuickPicksResponse

	if err := json.NewDecoder(response.Body).Decode(&quickPicks); err != nil {
		return nil, err
	}

	return &quickPicks, nil
}
